File: Java/aula_4/Aula_4_Complexo.java
package aula_4;
public class Aula_4_Complexo {
    private int real, imaginario;

    public Aula_4_Complexo() {
        real = 0;
        imaginario = 0;
    }

    public Aula_4_Complexo(int r) {
        real = r;
        imaginario = 0;
    }

    public Aula_4_Complexo(int r, int i) {
        real = r;
        imaginario = i;
    }

    public void inicializaNumero(int r, int i) {
        real = r;
        imaginario = i;
    }

    public void imprimirNumero() {
        if (imaginario > 0)
            System.out.println("Numero Complexo: " + real + " + " + imaginario + "i");
        else if (imaginario == 0)
            System.out.println(real);
        else
            System.out.println(real + " " + imaginario + "i");
    }
 
    public boolean elgual(Aula_4_Complexo c) {
        if (c.real == this.real && c.imaginario == this.imaginario)
            return true;
        else
            return false;
    }

    public void soma(Aula_4_Complexo c) {
        this.real += c.real;
        this.imaginario += c.imaginario;
    }

    public void subtrai(Aula_4_Complexo c) {
        this.real -= c.real;
        this.imaginario -= c.imaginario;
    }

    public void multiplica(Aula_4_Complexo c) {
        int aux_r = (this.real * c.real) - (this.imaginario * c.imaginario);
        int aux_i = (this.real * c.imaginario) + (this.imaginario * c.real);

        this.real = aux_r;
        this.imaginario = aux_i;
    }

    public void divide(Aula_4_Complexo c) {
        int aux_r = ((this.real * c.real) + (this.imaginario * c.imaginario)) / ((c.real * c.real) + (c.imaginario * c.imaginario));
        int aux_i = ((this.imaginario * c.real) - (this.real * c.imaginario)) / ((c.real * c.real) + (c.imaginario * c.imaginario));
        this.real = aux_r;
        this.imaginario = aux_i;
    }
}

File: Python/Aula/Aula_4_Complexo.py
class Complexo:
    def __init__(self, real=1, imaginario=1):
        self.real = real
        self.imaginario = imaginario

    def inicializarComplexo(self, real, imaginario):
        self.real = real
        self.imaginario = imaginario

    def imprimirComplexo(self):
        if (self.imaginario > 0):
            print(f"Complexo: {self.real} + {self.imaginario}i")
        elif (self.imaginario == 0):
            print(f"Complexo: {self.real}")
        else:
            print(f"Complexo: {self.real} {self.imaginario}i")

    def elgual(self, complexo):
        return (complexo.real == self.real and complexo.imaginario == self.imaginario)

    def soma(self, c):
        self.real += c.real
        self.imaginario += c.imaginario

    def subtrai(self, c):
        self.real -= c.real
        self.imaginario -= c.imaginario

    def multiplica(self, c):
        aux_r = (self.real * c.real) - (self.imaginario * c.imaginario)
        aux_i = (self.real * c.imaginario) + (self.imaginario * c.real)
        self.real = aux_r
        self.imaginario = aux_i

    def divide(self, c):
        aux_r = ((self.real * c.real) + (self.imaginario * c.imaginario)) / ((c.real * c.real) + (c.imaginario * c.imaginario))
        aux_i = ((self.imaginario * c.real) - (self.real * c.imaginario)) / ((c.real * c.real) + (c.imaginario * c.imaginario))
        self.real = aux_r
        self.imaginario = aux_i
